src/software/new_geom/util/closest_point.h

```diff
@@ -12,3 +13,15 @@
  */
 Point closestPointOnLine(const Point &p, const Line &l);
 Point closestPointOnLine(const Line &l, const Point &p);
+
+/**
+ * Finds the Point on line segment closest to point.
+ *
+ * @param p the point.
+ *
+ * @param segment the line segment.
+ *
+ * @return the Point on line segment closest to point.
+ */
+Point closestPointOnSeg(const Point &p, const Segment &segment);
+Point closestPointOnSeg(const Segment &segment, const Point &p);
```

Comment:
Not sure how I feel about this naming. In general we've just been naming functions `distance`, `intersection`, etc. and so this would be `closestPoint` following that trend. But the suffix is maybe helpful here? Also not sure we should both arg orderings here since this is implicitly directional.

@MathewMacDougall @akhilveeraghanta @EvanMorcom @jonathanlew for your thoughts

Reply:
I think closestPoint is too vague. I think it should be closestPointOnGeom in a closest_point_on_geom file.

Reply:
I think I prefer `closestPoint` because it matches the other naming styles and is sufficiently descriptive without providing redundant information about its args. I don't see how adding `OnGeom` to the name would be more helpful.

I also think `closestPoint` is _fairly_ bidirectional like distance. I think it would be ok because you will always have `Point` as one of the args, so it's pretty obvious you're getting the `closestPoint` to the given point on whatever the other arg is

Reply:
I like `closestPointOnSeg` more because it's more explicit. `closestPoint` would make me think "Closest to what", but that might be because I'm new to working with geom.

Reply:
I think `closestPoint` is more consistent with `distance` and `intersection` like Gareth mentioned, we don't do `IntersectionWithSegment`

If we do `closestPointOnGeom` does that mean every closest point function will get a separate file based on which "geom" type it is?